sdk/core/Azure.Core/src/AzureKeyCredential.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using System.Threading;
+
+namespace Azure.Core
```

Comment:
@pakrym, should it be here or in Azure root?

Reply:
I somewhat agree with https://github.com/Azure/azure-sdk-for-net/pull/10668#discussion_r394031804 but for consistency I would've kept it in Azure.Core.

Reply:
Everyone should be using DefaultAzureCredential anyway

Reply:
I just worry that if we keep it in Azure.Core, people will start adding Azure.Core namespace to their files. I think this type is simple enough and will be used in many services, and so Azure root  would make the experience better.

Reply:
@pakrym I don't believe it's the same case as TokenCredential. Perhaps TokenCredential should have been in Azure rather than Azure.Core, however a library user will very infrequently use the type. They will more likely use an implementation from Azure.Identity like DefaultAzureCredential. However this type they will have to reference directly.

Reply:
Interesting. I will leave it here for this PR, and will revisit the namespace.
Issue https://github.com/Azure/azure-sdk-for-net/issues/10672